ansible/roles/prep.dirs/tasks/repo.yml

```diff
@@ -9,25 +9,31 @@
   loop_control:
     loop_var: mock_config
 
-- name: adapt mock config to use additional template for ceph
+- name: adapt mock config to use additional dev template for ceph
   block:
     - name: determine ceph repo base url
       shell: curl -s "https://shaman.ceph.com/api/search/?project=ceph&distros=centos/{{ item }}/x86_64&flavor=default&ref=main&sha1=latest" | jq -r .[0].url
       register: get_ceph_repo_url
+      when: ceph_base_url is not defined
 
     - set_fact:
         "ceph_el{{ item }}_repo_base_url": "{{ get_ceph_repo_url.stdout }}"
+      when: ceph_base_url is not defined
```

Comment:
You could go for a nested block here to group both tasks when `ceph_base_url` is not defined.

Reply:
Good idea, I've made that change.